### PR TITLE
[FLASK] Update snap installation permission warning UI

### LINKED ISSUE #19490: Design audit: Warning modal
**Figma:** https://www.figma.com/file/Wlwk0cTx6PQkNhiX7vN13e/Snaps-Platform-v1?type=design&node-id=177%3A16851&t=kY4jMt2plPPUv0I8-1

- [ ] `48px` title icon
- [ ] `Heading-MD` / `18px` title
- [ ] `32px` margin between check list items
- [ ] `base` / `14px` font-size for list items
- [ ] `bold` -> `medium` font-weight for list items
- [ ] Update description copy to _Take a moment to review the permissions being requested. Only continue if you trust **Arbitrary Manager**._

### PR DESCRIPTION
**Pull request by david0xd**
## Explanation
Addressing UI update request.
Fixes: https://github.com/MetaMask/metamask-extension/issues/19490

Design specification: https://www.figma.com/file/Wlwk0cTx6PQkNhiX7vN13e/Snaps-Platform-v1?type=design&node-id=117-17974&t=UUGLiKRpDlpuWdrk-0

### After
Screenshot 2023-06-07 at 18 03 21
Screenshot 2023-06-07 at 18 04 07

### Before
Screenshot 2023-06-07 at 17 41 54
Screenshot 2023-06-07 at 17 42 38

## Manual Testing Steps
Try to install a snap which has permissions that would trigger a warning popup to appear. 
`npm:@metamask/test-snap-bip32` is a good example and can be found here: https://metamask.github.io/test-snaps/latest/

## Pre-merge author checklist

- [ ] I've clearly explained:
  - [ ] What problem this PR is solving
  - [ ] How this problem was solved
  - [ ] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: ui/pages/permissions-connect/snaps/snap-install/snap-install.js [hunk 1]
```diff
@@ -179,6 +179,7 @@ export default function SnapInstall({
           onCancel={() => setIsShowingWarning(false)}
           onSubmit={onSubmit}
           warnings={warnings}
+          snapName={snapName}
```
Comment (FrederikBolding) [inline ui/pages/permissions-connect/snaps/snap-install/snap-install.js:182]: Please update this on `snap-update` too!

**Review by FrederikBolding — CHANGES_REQUESTED** [review]

**Comment by eriknson** [comment]
Design lgtm!

**Comment by eriknson** [comment]
Implementation / design lgtm!

File: ui/pages/permissions-connect/snaps/snap-install/snap-install.js (hunk 1 above)
Comment (david0xd) [reply ui/pages/permissions-connect/snaps/snap-install/snap-install.js:182]: Updated.

**Review by FrederikBolding — APPROVED** [review]

**Review by Mrtenz — APPROVED** [review]

**Pull request merged**